src/sage/combinat/tableau.py

```diff
@@ -8183,25 +8182,20 @@ def random_element(self):
         t = [[None] * n for n in p]
 
         # Get the cells in the Young diagram
-        cells = []
-        for i in range(len(p)):
-            for j in range(p[i]):
-                cells.append((i, j))
+        cells = [(i, j) for i in range(len(p)) for j in range(p[i])]
 
         m = sum(p)
-        while m > 0:
+        while m:
```

Comment:
Unless this should be `while m != 0`, I think `m > 0` is more readable.

Reply:
Well,  the boolean test is supposed to be faster.

Reply:
Fair enough, as long as `m` can never be negative (which I think is true here so that's fine).